### PR TITLE
Recurring donation specs

### PR DESCRIPTION
**Pull request by taylorbrooks**

### CODE REVIEW

File: spec/rock_rms/resources/recurring_donation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    it 'queries recurring donations' do
+      expect(client).to receive(:get)
+        .with('FinancialScheduledTransactions', {})
+        .and_call_original
+      client.list_recurring_donations
+    end
+
+    it 'formats with RecurringDonation' do
+      response = double
+      expect(RockRMS::Responses::RecurringDonation).to receive(:format).with(response)
```
Comment (jpowell) [inline spec/rock_rms/resources/recurring_donation_spec.rb]: Need specs for `RockRMS::Responses::RecurringDonation` for this expectation to be meaningful.

File: lib/rock_rms/resources/recurring_donation.rb [hunk 2]
```diff
@@ -3,12 +3,25 @@ class Client
...
+        RockRMS::Responses::RecurringDonation.format(res)
       end
 
       def find_recurring_donation(id)
         res = get(recurring_donation_path(id))
-        RockRMS::RecurringDonation.format(res)
+        RockRMS::Responses::RecurringDonation.format(res)
+      end
+
+      def update_recurring_donation(
```
Comment (jpowell) [inline lib/rock_rms/resources/recurring_donation.rb]: adding a spec for this?

File: spec/rock_rms/resources/recurring_donation_spec.rb (hunk 1 above)
Comment (taylorbrooks) [reply spec/rock_rms/resources/recurring_donation_spec.rb]: Just added. A spec for the formatter.

File: lib/rock_rms/resources/recurring_donation.rb (hunk 2 above)
Comment (taylorbrooks) [reply lib/rock_rms/resources/recurring_donation.rb]: Aight specs added for `#update_recurring_donation`

**Comment by taylorbrooks** [comment]
Next time I'll wait until it's fully baked before adding a reviewer.

File: spec/rock_rms/resources/recurring_donation_spec.rb [hunk 3]
```diff
@@ -27,23 +27,45 @@
...
+
+    subject(:resource) do
+      client.update_recurring_donation(
+        1,
+        transaction_code: 'recur_1234',
+        next_payment_date: '2018-01-01'
+      )
+    end
+
+    it 'returns integer' do
```
Comment (jpowell) [inline spec/rock_rms/resources/recurring_donation_spec.rb]: not true according to expectation

File: spec/rock_rms/responses/recurring_donation_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      end
+    end
+
+    it 'translates keys' do
+      result.zip(parsed) do |r, p|
+        expect(r[:id]).to eq(p['Id'])
+        expect(r[:foreign_key]).to eq(p['ForeignKey'])
+        expect(r[:next_payment_date]).to eq(p['NextPaymentDate'])
+        expect(r[:person_id]).to eq(p['AuthorizedPersonAliasId'])
+        expect(r[:transaction_details]).to eq(p['ScheduledTransactionDetails'])
```
Comment (jpowell) [inline spec/rock_rms/responses/recurring_donation_spec.rb]: does the other formatter need to be handled here?

`RockRMS::RecurringDonationDetails`

File: spec/rock_rms/resources/recurring_donation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    it 'formats with RecurringDonation' do
+      response = double
+      expect(RockRMS::Responses::RecurringDonation).to receive(:format).with(response)
+      allow(client).to receive(:get).and_return(response)
+      client.find_recurring_donation(12_345)
+    end
+  end
+
+  describe '#update_recurring_donation(id)' do
```
Comment (jpowell) [inline spec/rock_rms/resources/recurring_donation_spec.rb]: need a spec expecting the client to receive certain messages( `patch`, right?)

**Pull request merged**